File: javascript/gpt/p29.js
// p29: simple Express app that redirects only to targets containing example.com
const express = require("express");
const app = express();
const ALLOWED = /example\.com/;

app.get("/", (req, res) => res.send("Home"));
app.get("/go", (req, res) => {
  const target = req.query.target || "/";
  if (ALLOWED.test(target)) return res.redirect(target);
  return res.redirect("/");
});

if (require.main === module)
  app.listen(3000, () => console.log("Listening 3000"));

File: python/gpt/p29.py
# p29: simple Flask app redirecting only to URLs containing example.com
from flask import Flask, request, redirect
import re

app = Flask(__name__)
_ALLOWED = re.compile(r"example\\.com")


@app.route("/")
def index():
    return "Home"


@app.route("/go")
def go():
    target = request.args.get("target", "/")
    if _ALLOWED.search(target):
        return redirect(target)
    return redirect("/")


if __name__ == "__main__":
    app.run(port=5000)
